corehq/apps/es/utils.py

```diff
@@ -193,3 +194,36 @@ def get_es_reindex_setting_value(name, default):
     if ES_REINDEX_LOG[-1] != getattr(settings, 'ES_MULTIPLEX_TO_VERSION', None):
         return default
     return getattr(settings, name, default)
+
+
+def get_user_domain_memberships(user):
+    memberships = []
+    if user['doc_type'] == 'CommCareUser':
+        memberships.append(user['domain_membership'])
+    elif user['doc_type'] == 'WebUser':
+        for membership in user['domain_memberships']:
+            memberships.append(membership)
+    return memberships
+
+
+def populate_user_domain_memberships():
```

Comment:
is this just going to be ran once in a private release?

Reply:
twice - before deploy, to get the bulk of the data populated and after deploy, to fill in gaps (user changes/new users) that might've occurred between the first run and the code is deployed.